schematic_api/api/routes.py

```diff
@@ -206,130 +220,193 @@ def get_temp_jsonld(schema_url):
-                    app.logger.warning(f'Currently we do not support returning multiple files as Excel format at once. Only {t} would get returned. ')
-                return result
-
-    return all_results
+    @staticmethod
+    def get_manifests_route(schema_url: str, data_type:List[str], use_annotations: Optional[str]=None, dataset_id: Optional[List[str]]=None, title: Optional[str]=None, asset_view: Optional[str]=None, output_format: Optional[str]="google_sheet") -> str|pd.DataFrame|BinaryIO:
+        """Generate a new manifest template or create an existing manifest in google sheet/excel/dataframe format. 
+
+        Args:
+            schema_url (str): data model in json ld format.
+            output_format (str, optional): output format of manifests. Available options are: google sheet, excel, and dataframe. Defaults to "google_sheet".
+            use_annotations (bool, optional): whether to use existing annotations during manifest generation. Defaults to False
+            dataset_id (list[str], optional): id of a given dataset. Defaults to None.
+            data_type (list[str], optional):a list of data types. Defaults to None.
+            asset_view (str, optional): id of a file view. Defaults to None.
+            title (str, optional): title of the new manifest. Defaults to None.
+
+        Raises:
+            ValueError: could not submit data_type = "all manifests" and dataset id at the same time
+            ValueError: do not enter multiple dataset ids if calling "all manifests" for data_type.
+
+        Returns:
+            str|pd.DataFrame|BinaryIO: returns google sheet url, pandas dataframe, or excel file based on output_format parameter
+        """
+        # initalize manifest generation class
+        access_token=request.headers.get('X-Auth')
+        mg = ManifestGeneration(schema_url=schema_url, access_token=access_token, output_format=output_format, title=title, use_annotations=use_annotations, dataset_id=dataset_id, data_type=data_type, asset_view=asset_view)
+
+        # load configuration file
+        mg._load_config_(app)
+
+        # get path to temporary JSON-LD file
+        jsonld = get_temp_jsonld(schema_url)
+
+        # Gather all returned result urls
+        if data_type[0] == 'all manifests':
+            if dataset_id:
+                raise ValueError('Do not enter multiple dataset ids if calling "all manifests" for data_type.')
+            sg = SchemaGenerator(path_to_json_ld=jsonld)
+            component_digraph = sg.se.get_digraph_by_edge_type('requiresComponent')
+            components = list(component_digraph.nodes())
+        else:
+            components = data_type
+
+        # when output_format = "google_sheet" or output_format = "data_frame", a list of outputs get returned if needed
+        all_outputs = mg.generate_manifest_and_collect_outputs(access_token=access_token, data_type_lst=components, dataset_id=dataset_id)
+
+        if all_outputs:
```

Comment:
Doesn't this `if` condition open up the possibility that the return value is `None`, which isn't one of the return types?